### PR TITLE
imageio_avif: make legacy overrides AVIF specific

### PR DESCRIPTION
**Pull request by kmilos**
Reflects 8e3f1271e and d13dcafd4a75fa3405 a bit better

This change restricts the workaround specifically to AVIFs only since the color profile parsing now supports HEIFs as well.

### CODE REVIEW

File: src/common/imageio_avif.c [hunk 1]
```diff
@@ -230,6 +230,35 @@ int dt_imageio_avif_read_profile(const char *filename, uint8_t **out, dt_colorsp
...
+         && avif_image.matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_BT709)
+      {
+        /* must be actual Rec. 709 instead of 2.2 gamma*/
+        cicp->transfer_characteristics = AVIF_TRANSFER_CHARACTERISTICS_BT709;
+        over =  true;
+      }
+
+      if(over)
+      {
+        dt_print(DT_DEBUG_IMAGEIO, "Overriding nclx color profile for AVIF image [%s]: 1/%d/%d to 1/%d/%d\n",
```
Comment (TurboGit) [inline src/common/imageio_avif.c:257]: we usually add some tag before the string (and no capital letter), so should be:

`"[imageio_avif_read_profile] overriding ..."`

Comment (kmilos) [reply src/common/imageio_avif.c:257]: Yes, I noticed that as well, but kept the style consistent with the rest of this source file on purpose... If this is the only comment, maybe we can go ahead w/ the merge and we do another PR where we change all the strings in this file, or I can add another commit to this PR doing that?

Comment (TurboGit) [reply src/common/imageio_avif.c:257]: Adding another commit in this PR seems the way to go to me. TIA

Comment (kmilos) [reply src/common/imageio_avif.c]: Ok, now styled after imageio_j2k.c, imageio_png.c, and imageio_tiff.c

File: src/common/imageio_avif.c [hunk 2]
```diff
@@ -230,10 +217,43 @@ int dt_imageio_avif_read_profile(const char *filename, uint8_t **out, dt_colorsp
     cicp->color_primaries = avif_image.colorPrimaries;
     cicp->transfer_characteristics = avif_image.transferCharacteristics;
     cicp->matrix_coefficients = avif_image.matrixCoefficients;
+
+    /* fix up mistagged legacy AVIFs */
+    if(avif_image.colorPrimaries == AVIF_COLOR_PRIMARIES_BT709)
+    {
+      bool over = false;
```
Comment (TurboGit) [inline src/common/imageio_avif.c:224]: sorry did not spot that before, you need to use gboolean (bool will break on some platforms) and FALSE.

File: src/common/imageio_avif.c [hunk 3]
```diff
@@ -230,10 +217,43 @@ int dt_imageio_avif_read_profile(const char *filename, uint8_t **out, dt_colorsp
...
+      /* mistagged sRGB AVIFs exported before dt 3.8 */
+      if(avif_image.transferCharacteristics == AVIF_TRANSFER_CHARACTERISTICS_SRGB
+         && avif_image.matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_BT709)
+      {
+        /* must be code value 5 (IEC 61966-2-1 sYCC) */
+        cicp->matrix_coefficients = AVIF_MATRIX_COEFFICIENTS_BT470BG;
+        over =  true;
```
Comment (TurboGit) [inline src/common/imageio_avif.c:231]: TRUE

File: src/common/imageio_avif.c [hunk 4]
```diff
@@ -230,10 +217,43 @@ int dt_imageio_avif_read_profile(const char *filename, uint8_t **out, dt_colorsp
...
+      }
+      /* mistagged Rec. 709 AVIFs exported before dt 3.6 */
+      else if(avif_image.transferCharacteristics == AVIF_TRANSFER_CHARACTERISTICS_BT470M
+         && avif_image.matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_BT709)
+      {
+        /* must be actual Rec. 709 instead of 2.2 gamma*/
+        cicp->transfer_characteristics = AVIF_TRANSFER_CHARACTERISTICS_BT709;
+        over =  true;
```
Comment (TurboGit) [inline src/common/imageio_avif.c:239]: TRUE

**Review by TurboGit — CHANGES_REQUESTED** [review]
Almost ok !

File: src/common/imageio_avif.c (hunk 2 above)
Comment (kmilos) [reply src/common/imageio_avif.c:224]: Done

Comment (TurboGit) [reply src/common/imageio_avif.c:224]: I don't see any change! A missing push?

Comment (kmilos) [reply src/common/imageio_avif.c:224]: Argh, forced pushed an empty commit, forgot to stage 🤦

**Review by TurboGit — APPROVED** [review]
Thanks !

**Pull request merged**